module/observationSubmissions/helper.js

```diff
@@ -126,11 +127,24 @@ module.exports = class ObservationSubmissionsHelper {
                   ["name","description"],
               );
 
+              programDocument = programDocument.data.data;
+
               if( programDocument[0] ) {
                 observationSubmissionsDocument['programInfo'] = programDocument[0];
 
               }
 
+            }
+
+                
+            let entityTypeDocumentsAPICall = await entityManagementService.entityTypeDocuments(
+              {"name":observationSubmissionsDocument.entityType}
+            );
+            
+            if(entityTypeDocumentsAPICall.success){
```

Comment:
@borkarsaish65 handle the error here too
 if (entityTypeResponse?.success && entityTypeResponse?.data?.length) {
            observationSubmissionsDocument.entityTypeId = entityTypeResponse.data[0]._id;
        }